pkg/sources/github/github.go

```diff
@@ -742,72 +797,37 @@ func (s *Source) getReposByUser(ctx context.Context, user string) ([]string, err
 			continue
 		}
 		if err != nil {
-			return nil, fmt.Errorf("could not list repos for user %s: %w", user, err)
+			return fmt.Errorf("could not list repos for user %s: %w", user, err)
 		}
 		if res == nil {
 			break
 		}
 
 		logger.V(2).Info("Listed repos", "page", opts.Page, "last_page", res.LastPage)
 		for _, r := range someRepos {
-			if s.ignoreRepo(r.GetFullName()) {
-				continue
-			}
-			if !s.includeRepo(r.GetFullName()) {
+			if r.GetFork() && !s.conn.IncludeForks {
 				continue
 			}
 
-			if r.GetFork() && !s.conn.IncludeForks {
+			repo, err := s.normalizeRepo(r.GetCloneURL())
+			if err != nil {
+				s.log.V(2).Info("could not normalize repo", "repo", r.GetFullName(), "error", err)
 				continue
 			}
-			repos = append(repos, r.GetCloneURL())
+			s.filteredRepoCache.Set(r.GetFullName(), repo)
 		}
 		if res.NextPage == 0 {
 			break
 		}
 		opts.Page = res.NextPage
 	}
-	return repos, nil
-}
-
-func (s *Source) includeRepo(r string) bool {
-	if len(s.includeRepos) == 0 {
-		return true
-	}
 
-	logger := s.log.WithValues("repo", r)
-	for _, include := range s.includeRepos {
-		g, err := glob.Compile(include)
-		if err != nil {
-			logger.V(2).Info("invalid glob", "glob", include, "error", err)
-			continue
-		}
-		if g.Match(r) {
-			logger.V(2).Info("including repo")
-			return true
-		}
-	}
-	return false
-}
-
-func (s *Source) ignoreRepo(r string) bool {
-	logger := s.log.WithValues("repo", r)
-	for _, ignore := range s.ignoreRepos {
-		g, err := glob.Compile(ignore)
```

Comment:
does this recompile every glob pattern every time we find a repo?? that seems really inefficient!

Reply:
Yep, not good. Updated it to online compile on init.